dotenv-linter/src/schema.rs

```diff
@@ -0,0 +1,335 @@
+use anyhow::Result;
+use serde::Deserialize;
+use std::{collections::HashMap, fs::File, io::BufReader, path::Path};
+
+#[derive(Deserialize, Default)]
+#[serde(default)]
+struct ReadDotEnvSchema {
+    pub version: String,
+    pub allow_other_keys: bool,
+    pub entries: Vec<SchemaEntry>,
+}
+
+pub struct DotEnvSchema {
+    pub version: String,
+    pub allow_other_keys: bool,
+    pub entries: HashMap<String, SchemaEntry>,
+}
+#[derive(Deserialize, Default)]
+#[serde(default)]
+pub struct SchemaEntry {
+    pub key: String,
+    pub required: bool,
+    #[serde(rename = "type")]
+    pub value_type: SchemaValueType,
+    pub regex: Option<String>,
+}
+#[derive(Deserialize, Default)]
+
+pub enum SchemaValueType {
+    #[default]
+    String,
+    Number,
```

Comment:
Btw, I think it makes sense to be more precise for number representation here. 
E.g be able to validate integer against floating point number